index.js

```diff
@@ -3,81 +3,84 @@ const addButton = document.querySelector('#add');
 const titleInput = document.querySelector('#title');
 const autherInput = document.querySelector('#author');
 
-// Create a collection that keeps a list of books
-const listOfBooks = [
-  {
-    title: 'atomic habit',
-    author: 'James Clear',
-  },
-  {
-    title: 'Rich dad poor dad',
-    author: 'Robert kiosaki',
-  },
-  {
-    title: 'Think and grow rich',
-    author: 'napleon hill',
-  },
-];
+// put variables and most used functions first to acces them
+let booksData = [];
+let availableStorage;
+let count = 0;
+
+function storeData(booksData) {
+  if (availableStorage) {
+    const jsonData = JSON.stringify(booksData);
+    availableStorage.setItem('books', jsonData);
+  }
+}
+
+function retrieveData() {
+  const bookData = availableStorage.getItem('books');
+  const parseBookData = JSON.parse(bookData);
+  if (parseBookData) {
+    booksData = parseBookData;
+  }
+}
 
-// Create a function to add a new book to the collection, with title and author
-function addBook(title, author) {
-  const tempBook = {
-    title,
-    author,
-  };
-  // save the new added book into the collection
-  listOfBooks.push(tempBook);
-  // desplay the new added book into the page
-  const addedBooks = document.querySelector('#add_books');
+// define class with object constructor
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
+
+  static addBook(book) {
+    const newBook = new Book(book.title, book.author);
+    booksData.push(newBook);
+    storeData(booksData);
+    window.location.reload();
+  }
+
+  static removeBook(book) {
+    booksData.splice(
+      booksData.findIndex((e) => e.title === book.title
+      && e.author === book.author),
+    );
+    storeData(booksData);
+  }
+}
+
```

Comment:
- [ ] it would be better to solve the bug in the following video when I remove the first item and reload it come back and another one got deleted 
https://www.loom.com/share/71ba7ff77f244839a1314f5a08aec7d8